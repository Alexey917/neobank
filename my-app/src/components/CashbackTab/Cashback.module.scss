@use '../../styles/App' as *;

.cashback {
  @include flex-properties(flex, row, null, null, 40px);
  flex-wrap: wrap;
  margin-top: 49px;

  &__wrapper {
    flex-basis: 406px;
    padding: 37px 32px 32px 32px;
    background-color: rgba(234, 236, 238, 1);
    box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.2),
      0px 0px 8px 0px rgba(0, 0, 0, 0.08);

    border-radius: 28px;
  }

  &__wrapper_1,
  &__wrapper_3,
  &__wrapper_5 {
    background-color: rgba(136, 179, 184, 0.6);
  }

  &__title {
    @include text-properties(
      Ubuntu,
      16px,
      147%,
      null,
      500,
      rgba(67, 67, 67, 0.9)
    );
    margin: 0 0 22px 0;
  }

  &__text {
    @include text-properties(Ubuntu, 36px, 147%, null, 700, rgba(0, 0, 0, 1));
    margin: 0;
  }
}

@media (max-width: 1300px) {
  .cashback {
    @include flex-properties(flex, row, center, null, 40px);
  }
}
