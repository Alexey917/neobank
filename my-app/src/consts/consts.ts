import { isOver18 } from '../utils/isOver18';
import money from '../assets/sprite.svg';
import calendar from '../assets/sprite.svg';
import clock from '../assets/sprite.svg';
import bag from '../assets/sprite.svg';
import card from '../assets/sprite.svg';
import { TCustomizeFormData } from '../types/types';

export const DATA_FORM: TCustomizeFormData = [
  {
    field: 'amount',
    valueAsNumber: true,
    errors: {
      required: {
        value: true,
        message: 'Enter amount',
      },
      min: {
        value: 15000,
        message: 'At least 15,000',
      },
      max: {
        value: 600000,
        message: 'No more than 600,000',
      },
    },
  },
  {
    label: 'Your last name',
    field: 'lastName',
    placeholder: 'For Example Doe',
    errors: {
      required: 'last name is required',
    },
  },
  {
    label: 'Your first name',
    field: 'firstName',
    placeholder: 'For Example Jhon',
    errors: {
      required: 'first name is required',
    },
  },
  {
    label: 'Your patronymic',
    field: 'middleName',
    placeholder: 'For Example Victorovich',
  },
  { label: 'Select term', field: 'term', placeholder: '6 month' },
  {
    label: 'Your email',
    field: 'email',
    placeholder: 'test@gmail.com',
    errors: {
      required: {
        value: true,
        message: 'Email is required',
      },
      pattern: {
        value:
          /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
        message: 'Invalid email address',
      },
    },
  },
  {
    label: 'Your date of birth',
    field: 'birthdate',
    placeholder: 'Select Date and Time',
    errors: {
      required: {
        value: true,
        message: 'Date of birth is required',
      },
      pattern: {
        value: /^(19|20)\d{2}-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])$/,
        message: 'Please use format YYYY-MM-DD',
      },
      validate: {
        isOver18: (value: string) =>
          isOver18(value) || 'You must be over 18 years old',
      },
    },
  },
  {
    label: 'Your passport series',
    field: 'passportSeries',
    placeholder: '0000',
    errors: {
      required: {
        value: true,
        message: 'Passport series is required',
      },
      minLength: {
        value: 4,
        message: 'The series must consist of 4 digits',
      },
      maxLength: {
        value: 4,
        message: 'The series must consist of 4 digits',
      },
    },
  },
  {
    label: 'Your passport number',
    field: 'passportNumber',
    placeholder: '000000',
    errors: {
      required: {
        value: true,
        message: 'Passport number is required',
      },
      minLength: {
        value: 6,
        message: 'The number must consist of 6 digits',
      },
      maxLength: {
        value: 6,
        message: 'The number must consist of 6 digits',
      },
    },
  },
];

export const RECEIVE_CARD = [
  {
    question: 'How to get a card?',
    answer:
      'We will deliver your card by courier free of charge. Delivery in Moscow and St. Petersburg - 1-2 working days. For other regions of the Russian Federation - 2-5 working days.',
  },

  {
    question:
      'What documents are needed and how old should one be to get a card?',
    answer: 'Need a passport. You must be between 20 and 70 years old.',
  },

  {
    question: 'In what currency can I issue a card?',
    answer: 'In rubles, dollars or euro',
  },

  {
    question: 'How much income do I need to get a credit card?',
    answer:
      'To obtain a credit card, you will need an income of at least 25,000 rubles per month after taxes.',
  },

  {
    question: "How do I find out about the bank's decision on my application?",
    answer:
      'After registration, you will receive an e-mail with a decision on your application.',
  },
];

export const USING_CARD = [
  {
    question: 'What is an interest free credit card?',
    answer:
      'A credit card with a grace period is a bank card with an established credit limit, designed for payment, reservation of goods and services, as well as for receiving cash, which allows you to use credit funds free of charge for a certain period.',
  },

  {
    question: 'How to activate a credit card',
    answer:
      'You can activate your credit card and generate a PIN code immediately after receiving the card at a bank branch using a PIN pad.',
  },

  {
    question: 'What is a settlement date?',
    answer:
      'The settlement date is the date from which you can pay off the debt for the reporting period. The settlement date falls on the first calendar day following the last day of the reporting period. The first settlement date is reported by the bank when transferring the issued credit card to the client, and then in the monthly account statement.',
  },

  {
    question: 'What do I need to know about interest rates?',
    answer:
      'For each reporting period from the 7th day of the previous month to the 6th day of the current month inclusive, a statement is generated for the credit card. The statement contains information on the amount and timing of the minimum payment, as well as the total amount of debt as of the date of issue.',
  },
];

export const RATES = [
  {
    title: 'Card currency',
    text: 'Rubles, dollars, euro',
  },

  {
    title: 'Interest free period',
    text: '0% up to 160 days',
  },

  {
    title: 'Payment system',
    text: 'Mastercard, Visa',
  },

  {
    title: 'Maximum credit limit on the card',
    text: '600 000 ₽',
  },

  {
    title: 'Replenishment and withdrawal',
    text: 'At any ATM. Top up your credit card for free with cash or transfer from other cards',
  },

  {
    title: 'Max cashback per month',
    text: '15 000 ₽',
  },

  {
    title: 'Transaction Alert',
    text: '60 ₽ — SMS or push notifications',
    textBr:
      '0 ₽ — card statement, information about transactions in the online bank',
  },
];

export const ABOUT_TAB = [
  {
    svg: money,
    svgId: 'money',
    title: 'Up to 50 000 ₽',
    text: 'Cash and transfers without commission and percent',
  },

  {
    svg: calendar,
    svgId: 'calendar',
    title: 'Up to 160 days',
    text: 'Without percent on the loan',
  },

  {
    svg: clock,
    svgId: 'clock',
    title: 'Free delivery',
    text: 'We will deliver your card by courier at a convenient place and time for you',
  },

  {
    svg: bag,
    svgId: 'bag',
    title: 'Up to 12 months',
    text: 'No percent. For equipment, clothes and other purchases in installments',
  },

  {
    svg: card,
    svgId: 'card',
    title: 'Convenient deposit and withdrawal',
    text: 'At any ATM. Top up your credit card for free with cash or transfer from other cards',
  },
];

export const CASHBACK_TAB = [
  {
    title: 'For food delivery, cafes and restaurants',
    text: '5%',
  },

  {
    title: 'In supermarkets with our subscription',
    text: '5%',
  },

  {
    title: "In clothing stores and children's goods",
    text: '2%',
  },

  {
    title: 'Other purchases and payment of services and fines',
    text: '1%',
  },

  {
    title: 'Shopping in online stores',
    text: 'up to 3%',
  },

  {
    title: 'Purchases from our partners',
    text: '30%',
  },
];
