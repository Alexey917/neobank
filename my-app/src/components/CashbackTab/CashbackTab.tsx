import { FC } from 'react';
import { CASHBACK_TAB } from '../../consts/consts';

import classes from './Cashback.module.scss';

export const CashbackTab: FC = () => {
  return (
    <section className={classes.cashback} aria-label="Cashback Terms">
      {CASHBACK_TAB.map((item, index) => (
        <article
          key={`cashback-${index}`}
          aria-labelledby={`cashback-title-${index}`}
          className={`${classes.cashback__wrapper} ${
            classes[`cashback__wrapper_${index}`]
          }`}
        >
          <h3 className={classes.cashback__title}>{item.title}</h3>
          <p className={classes.cashback__text}>{item.text}</p>
        </article>
      ))}
    </section>
  );
};
