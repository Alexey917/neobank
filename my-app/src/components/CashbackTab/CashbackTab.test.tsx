import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';

import { CashbackTab } from './CashbackTab';
import { CASHBACK_TAB } from '../../consts/consts';

describe('CashbackTab component', () => {
  it('CashbackTab render', () => {
    render(<CashbackTab />);
    CASHBACK_TAB.forEach((item) => {
      expect(screen.getByText(item.title)).toBeInTheDocument();
      if (item.text === '5%') {
        expect(screen.getAllByText(item.text));
      } else {
        expect(screen.getByText(item.text)).toBeInTheDocument();
      }
    });
  });

  it('renders without errors', () => {
    render(<CashbackTab />);
    expect(
      screen.getByRole('region', { name: 'Cashback Terms' }),
    ).toBeInTheDocument();
  });

  it('CashbackTab snapshot', () => {
    const CashbackSnapshot = render(<CashbackTab />);
    expect(CashbackSnapshot).toMatchSnapshot();
  });
});
