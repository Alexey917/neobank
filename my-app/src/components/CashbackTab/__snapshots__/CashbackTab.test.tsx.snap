// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AboutTab component > CashbackTab snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        aria-label="Cashback Terms"
        class="_cashback_afe6c5"
      >
        <article
          aria-labelledby="cashback-title-0"
          class="_cashback__wrapper_afe6c5 _cashback__wrapper_0_afe6c5"
        >
          <h3
            class="_cashback__title_afe6c5"
          >
            For food delivery, cafes and restaurants
          </h3>
          <p
            class="_cashback__text_afe6c5"
          >
            5%
          </p>
        </article>
        <article
          aria-labelledby="cashback-title-1"
          class="_cashback__wrapper_afe6c5 _cashback__wrapper_1_afe6c5"
        >
          <h3
            class="_cashback__title_afe6c5"
          >
            In supermarkets with our subscription
          </h3>
          <p
            class="_cashback__text_afe6c5"
          >
            5%
          </p>
        </article>
        <article
          aria-labelledby="cashback-title-2"
          class="_cashback__wrapper_afe6c5 _cashback__wrapper_2_afe6c5"
        >
          <h3
            class="_cashback__title_afe6c5"
          >
            In clothing stores and children's goods
          </h3>
          <p
            class="_cashback__text_afe6c5"
          >
            2%
          </p>
        </article>
        <article
          aria-labelledby="cashback-title-3"
          class="_cashback__wrapper_afe6c5 _cashback__wrapper_3_afe6c5"
        >
          <h3
            class="_cashback__title_afe6c5"
          >
            Other purchases and payment of services and fines
          </h3>
          <p
            class="_cashback__text_afe6c5"
          >
            1%
          </p>
        </article>
        <article
          aria-labelledby="cashback-title-4"
          class="_cashback__wrapper_afe6c5 _cashback__wrapper_4_afe6c5"
        >
          <h3
            class="_cashback__title_afe6c5"
          >
            Shopping in online stores
          </h3>
          <p
            class="_cashback__text_afe6c5"
          >
            up to 3%
          </p>
        </article>
        <article
          aria-labelledby="cashback-title-5"
          class="_cashback__wrapper_afe6c5 _cashback__wrapper_5_afe6c5"
        >
          <h3
            class="_cashback__title_afe6c5"
          >
            Purchases from our partners
          </h3>
          <p
            class="_cashback__text_afe6c5"
          >
            30%
          </p>
        </article>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      aria-label="Cashback Terms"
      class="_cashback_afe6c5"
    >
      <article
        aria-labelledby="cashback-title-0"
        class="_cashback__wrapper_afe6c5 _cashback__wrapper_0_afe6c5"
      >
        <h3
          class="_cashback__title_afe6c5"
        >
          For food delivery, cafes and restaurants
        </h3>
        <p
          class="_cashback__text_afe6c5"
        >
          5%
        </p>
      </article>
      <article
        aria-labelledby="cashback-title-1"
        class="_cashback__wrapper_afe6c5 _cashback__wrapper_1_afe6c5"
      >
        <h3
          class="_cashback__title_afe6c5"
        >
          In supermarkets with our subscription
        </h3>
        <p
          class="_cashback__text_afe6c5"
        >
          5%
        </p>
      </article>
      <article
        aria-labelledby="cashback-title-2"
        class="_cashback__wrapper_afe6c5 _cashback__wrapper_2_afe6c5"
      >
        <h3
          class="_cashback__title_afe6c5"
        >
          In clothing stores and children's goods
        </h3>
        <p
          class="_cashback__text_afe6c5"
        >
          2%
        </p>
      </article>
      <article
        aria-labelledby="cashback-title-3"
        class="_cashback__wrapper_afe6c5 _cashback__wrapper_3_afe6c5"
      >
        <h3
          class="_cashback__title_afe6c5"
        >
          Other purchases and payment of services and fines
        </h3>
        <p
          class="_cashback__text_afe6c5"
        >
          1%
        </p>
      </article>
      <article
        aria-labelledby="cashback-title-4"
        class="_cashback__wrapper_afe6c5 _cashback__wrapper_4_afe6c5"
      >
        <h3
          class="_cashback__title_afe6c5"
        >
          Shopping in online stores
        </h3>
        <p
          class="_cashback__text_afe6c5"
        >
          up to 3%
        </p>
      </article>
      <article
        aria-labelledby="cashback-title-5"
        class="_cashback__wrapper_afe6c5 _cashback__wrapper_5_afe6c5"
      >
        <h3
          class="_cashback__title_afe6c5"
        >
          Purchases from our partners
        </h3>
        <p
          class="_cashback__text_afe6c5"
        >
          30%
        </p>
      </article>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CashbackTab component > CashbackTab snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        aria-label="Cashback Terms"
        class="_cashback_afe6c5"
      >
        <article
          aria-labelledby="cashback-title-0"
          class="_cashback__wrapper_afe6c5 _cashback__wrapper_0_afe6c5"
        >
          <h3
            class="_cashback__title_afe6c5"
          >
            For food delivery, cafes and restaurants
          </h3>
          <p
            class="_cashback__text_afe6c5"
          >
            5%
          </p>
        </article>
        <article
          aria-labelledby="cashback-title-1"
          class="_cashback__wrapper_afe6c5 _cashback__wrapper_1_afe6c5"
        >
          <h3
            class="_cashback__title_afe6c5"
          >
            In supermarkets with our subscription
          </h3>
          <p
            class="_cashback__text_afe6c5"
          >
            5%
          </p>
        </article>
        <article
          aria-labelledby="cashback-title-2"
          class="_cashback__wrapper_afe6c5 _cashback__wrapper_2_afe6c5"
        >
          <h3
            class="_cashback__title_afe6c5"
          >
            In clothing stores and children's goods
          </h3>
          <p
            class="_cashback__text_afe6c5"
          >
            2%
          </p>
        </article>
        <article
          aria-labelledby="cashback-title-3"
          class="_cashback__wrapper_afe6c5 _cashback__wrapper_3_afe6c5"
        >
          <h3
            class="_cashback__title_afe6c5"
          >
            Other purchases and payment of services and fines
          </h3>
          <p
            class="_cashback__text_afe6c5"
          >
            1%
          </p>
        </article>
        <article
          aria-labelledby="cashback-title-4"
          class="_cashback__wrapper_afe6c5 _cashback__wrapper_4_afe6c5"
        >
          <h3
            class="_cashback__title_afe6c5"
          >
            Shopping in online stores
          </h3>
          <p
            class="_cashback__text_afe6c5"
          >
            up to 3%
          </p>
        </article>
        <article
          aria-labelledby="cashback-title-5"
          class="_cashback__wrapper_afe6c5 _cashback__wrapper_5_afe6c5"
        >
          <h3
            class="_cashback__title_afe6c5"
          >
            Purchases from our partners
          </h3>
          <p
            class="_cashback__text_afe6c5"
          >
            30%
          </p>
        </article>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      aria-label="Cashback Terms"
      class="_cashback_afe6c5"
    >
      <article
        aria-labelledby="cashback-title-0"
        class="_cashback__wrapper_afe6c5 _cashback__wrapper_0_afe6c5"
      >
        <h3
          class="_cashback__title_afe6c5"
        >
          For food delivery, cafes and restaurants
        </h3>
        <p
          class="_cashback__text_afe6c5"
        >
          5%
        </p>
      </article>
      <article
        aria-labelledby="cashback-title-1"
        class="_cashback__wrapper_afe6c5 _cashback__wrapper_1_afe6c5"
      >
        <h3
          class="_cashback__title_afe6c5"
        >
          In supermarkets with our subscription
        </h3>
        <p
          class="_cashback__text_afe6c5"
        >
          5%
        </p>
      </article>
      <article
        aria-labelledby="cashback-title-2"
        class="_cashback__wrapper_afe6c5 _cashback__wrapper_2_afe6c5"
      >
        <h3
          class="_cashback__title_afe6c5"
        >
          In clothing stores and children's goods
        </h3>
        <p
          class="_cashback__text_afe6c5"
        >
          2%
        </p>
      </article>
      <article
        aria-labelledby="cashback-title-3"
        class="_cashback__wrapper_afe6c5 _cashback__wrapper_3_afe6c5"
      >
        <h3
          class="_cashback__title_afe6c5"
        >
          Other purchases and payment of services and fines
        </h3>
        <p
          class="_cashback__text_afe6c5"
        >
          1%
        </p>
      </article>
      <article
        aria-labelledby="cashback-title-4"
        class="_cashback__wrapper_afe6c5 _cashback__wrapper_4_afe6c5"
      >
        <h3
          class="_cashback__title_afe6c5"
        >
          Shopping in online stores
        </h3>
        <p
          class="_cashback__text_afe6c5"
        >
          up to 3%
        </p>
      </article>
      <article
        aria-labelledby="cashback-title-5"
        class="_cashback__wrapper_afe6c5 _cashback__wrapper_5_afe6c5"
      >
        <h3
          class="_cashback__title_afe6c5"
        >
          Purchases from our partners
        </h3>
        <p
          class="_cashback__text_afe6c5"
        >
          30%
        </p>
      </article>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
